content/manuals/build/cache/garbage-collection.md

```diff
@@ -8,89 +8,266 @@ aliases:
 
 While [`docker builder prune`](/reference/cli/docker/builder/prune.md)
 or [`docker buildx prune`](/reference/cli/docker/buildx/prune.md)
-commands run at once, garbage collection runs periodically and follows an
-ordered list of prune policies.
+commands run at once, Garbage Collection (GC) runs periodically and follows an
+ordered list of prune policies. The BuildKit daemon clears the build cache when
+the cache size becomes too big, or when the cache age expires.
 
-Garbage collection runs in the BuildKit daemon. The daemon clears the build
-cache when the cache size becomes too big, or when the cache age expires. The
-following sections describe how you can configure both the size and age
-parameters by defining garbage collection policies.
+For most users, the default GC behavior is sufficient and doesn't require any
+intervention. Advanced users, particularly those working with large-scale
+builds, self-managed builders, or constrained storage environments, might
+benefit from customizing these settings to better align with their workflow
+needs. The following sections explain how GC works and provide guidance on
+tailoring its behavior through custom configuration.
 
-Each of the policy's parameters corresponds with a `docker buildx prune` command line
-argument. Details can be found in the 
-`docker buildx prune` [documentation](/reference/cli/docker/buildx/prune.md).
+## Garbage collection policies
+
+GC policies define a set of rules that determine how the build cache is managed
+and cleaned up. These policies include criteria for when to remove cache
+entries, such as the age of the cache, the amount of space being used, and
+other constraints like ensuring a minimum amount of free disk space or
+reserving a specific percentage of the disk for build cache.
+
+Each GC policy is evaluated in sequence, starting with the most specific
+criteria, and proceeds to broader rules if previous policies do not resolve the
```

Comment:
> and proceeds to broader rules if previous policies do not resolve the issue

What issue?

All policies will always run. The trick is to put more specific policies with lower limits as first rules. So eg. if your rules are
1) 48h 5GB
2) 10 GB

And the cache takes 7GB then the second rule will still run, but it will not match anything. And if your cache takes 11GB, first rule runs and deletes 2GB then the second rule also will not match anything because your storage is already 9GB when it gets called.

Reply:
Thanks, I like this example, I'll add in something like this.